Bug: Simulator deploy is broken

### Description
```
Container simulator-publisher  Starting
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: exec: "${BN_WORKDIR}/start.sh": stat ${BN_WORKDIR}/start.sh: no such file or directory: unknown

Execution failed for task ':simulator:startDockerContainer'.
> Process 'command 'sh'' finished with non-zero exit value 1
```
### Steps to Reproduce

try to deploy the blockstream simulator using helm or docker.

using the IDE, just go to `docker` tasks of gradle for project simulator and run `startDockerContainer` see output.

### Additional context

Is very likely that this was introduced on PR: https://github.com/hiero-ledger/hiero-block-node/pull/872

Also, once I was able to fix this issue, I encountered this that was due to changes in the `protobuf` project. but is only happening on deployment inside the container not on IDE direct Run manual.


```
at io.grpc.internal@1.72.0/io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:733)

	at io.grpc.internal@1.72.0/io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:714)

	at io.grpc.internal@1.72.0/io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)

	at io.grpc.internal@1.72.0/io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)

	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)

	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)

	at java.base/java.lang.Thread.run(Thread.java:1583)

Caused by: java.lang.IllegalStateException: Couldn't use Java reflection to implement protocol message reflection.

	at com.google.protobuf@4.30.2/com.google.protobuf.GeneratedMessage.invokeOrDie(GeneratedMessage.java:1962)

	at com.google.protobuf@4.30.2/com.google.protobuf.GeneratedMessage.access$1200(GeneratedMessage.java:46)

	at com.google.protobuf@4.30.2/com.google.protobuf.GeneratedMessage$FieldAccessorTable$RealOneofAccessor.has(GeneratedMessage.java:2229)

	at com.google.protobuf@4.30.2/com.google.protobuf.GeneratedMessage.hasOneof(GeneratedMessage.java:234)

	at com.google.protobuf@4.30.2/com.google.protobuf.GeneratedMessage.getAllFieldsMutable(GeneratedMessage.java:155)

	at com.google.protobuf@4.30.2/com.google.protobuf.GeneratedMessage.getAllFields(GeneratedMessage.java:217)

	at com.google.protobuf@4.30.2/com.google.protobuf.TextFormat$Printer.printMessage(TextFormat.java:745)

	at com.google.protobuf@4.30.2/com.google.protobuf.TextFormat$Printer.print(TextFormat.java:354)

	at com.google.protobuf@4.30.2/com.google.protobuf.TextFormat$Printer.print(TextFormat.java:339)

	at com.google.protobuf@4.30.2/com.google.protobuf.TextFormat$Printer.printToString(TextFormat.java:633)

	at com.google.protobuf@4.30.2/com.google.protobuf.AbstractMessage.toString(AbstractMessage.java:91)

	at org.hiero.block.simulator@0.11.0-SNAPSHOT/org.hiero.block.simulator.grpc.impl.PublishStreamObserver.onNext(PublishStreamObserver.java:78)

	at org.hiero.block.simulator@0.11.0-SNAPSHOT/org.hiero.block.simulator.grpc.impl.PublishStreamObserver.onNext(PublishStreamObserver.java:23)

	at io.grpc.stub@1.72.0/io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onMessage(ClientCalls.java:551)

	at io.grpc.internal@1.72.0/io.grpc.internal.DelayedClientCall$DelayedListener.onMessage(DelayedClientCall.java:473)

	at io.grpc.internal@1.72.0/io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1MessagesAvailable.runInternal(ClientCallImpl.java:661)

	at io.grpc.internal@1.72.0/io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1MessagesAvailable.runInContext(ClientCallImpl.java:648)

	... 5 more

Caused by: java.lang.IllegalAccessException: class com.google.protobuf.GeneratedMessage (in module com.google.protobuf) cannot access class org.hiero.block.api.protoc.PublishStreamResponse (in module org.hiero.block.protobuf) because module org.hiero.block.protobuf does not export org.hiero.block.api.protoc to module com.google.protobuf

	at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:394)

	at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:714)

	at java.base/java.lang.reflect.Method.invoke(Method.java:571)

	at com.google.protobuf@4.30.2/com.google.protobuf.GeneratedMessage.invokeOrDie(GeneratedMessage.java:1960)

	... 21 more


2025-05-21 01:10:39.370+0000 SEVERE  [org.hiero.block.simulator.grpc.impl.PublishStreamGrpcClientImpl streamBlock] Not allowed to send next batch of block items
```

--- a/protobuf/src/main/java/module-info.java
+++ b/protobuf/src/main/java/module-info.java
@@ -97,4 +97,7 @@ module org.hiero.block.protobuf {
     requires org.antlr.antlr4.runtime;
     requires static com.github.spotbugs.annotations;
     requires static java.annotation;
+
+    opens org.hiero.block.api.protoc to
+            com.google.protobuf;
 }

--- a/simulator/docker/Dockerfile
+++ b/simulator/docker/Dockerfile
@@ -5,6 +5,7 @@ ARG UNAME=hedera
 ARG UID=2000
 ARG GID=2000
 ARG BN_WORKDIR="/opt/hiero/block-node"
+ENV BN_WORKDIR=${BN_WORKDIR}
 RUN groupadd --gid $GID $UNAME
 RUN useradd --no-user-group --create-home --uid $UID --gid $GID --shell /bin/bash hedera
 WORKDIR ${BN_WORKDIR}
@@ -29,4 +30,4 @@ RUN SIMULATOR_DIR=$(ls -d simulator-*/) && \
     echo "#!/bin/bash\n${BN_WORKDIR}/${SIMULATOR_DIR}bin/simulator" > ${BN_WORKDIR}/start.sh && \
     chmod +x ${BN_WORKDIR}/start.sh
 
-ENTRYPOINT ["${BN_WORKDIR}/start.sh"]
+ENTRYPOINT ["sh","-c","${BN_WORKDIR}/start.sh"]

